src/main/java/rx/internal/operators/OperatorDoOnEmpty.java

```diff
@@ -0,0 +1,59 @@
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Subscriber;
+import rx.exceptions.Exceptions;
+import rx.functions.Action0;
+
+public final class OperatorDoOnEmpty<T> implements Observable.Operator<T, T> {
+
+    private final Action0 onEmpty;
+
+    public OperatorDoOnEmpty(Action0 onEmpty) {
+        this.onEmpty = onEmpty;
+    }
+
+    @Override
+    public Subscriber<? super T> call(final Subscriber<? super T> child) {
+
+        return new Subscriber<T>(child) {
+
+            private boolean isEmpty = true;
+            private boolean done = false;
+
+            @Override
+            public void onCompleted() {
+                if (done) {
+                    return;
+                }
+                if (isEmpty) {
+                    try {
+                        onEmpty.call();
+                    } catch (Throwable e) {
+                        Exceptions.throwOrReport(e,this);
+                    }
```

Comment:
add `return;` after call to `Exceptions.throwOrReport`. Thomas I suggest you read the comments on the github site because it seems like you are reacting to truncated comments (perhaps email notifications).